### DISCUSSION TITLE
💖 Vote for which articles I should write next!

### DISCUSSION THREAD

**Discussion by pretzelhammer** [pretzelhammer/rust-blog#26] [General]
I have a headline-driven style for writing articles. I think of a cool headline and then I try to write the article for it. The headline is arguably the most important part since it's what gets people to click on the article in the first place and read it. If the article has a boring headline nobody even gives it a chance.

Here are some headlines I'm thinking of writing articles for, please reply and tell me which grab your attention the most and which you'd be the most excited to read:

- A Spoonful of Sugar Helps the Rust Go Down
- Async Programming in Rust
- Beginner's Guide to Unsafe Rust
- Closures in Rust
- Exploring Function Signatures in Rust
- JavaScript Dev Learns Rust
- Learning Rust in 2021
- Learn Rust in an Hour
- Learn Rust Real Good Real Fast
- Learn Rust with Generative Art
- Macros in Rust
- Polymorphism in Rust
- Rust: Behavior Oriented Programming
- Rust Crash Course
- Rust: Declarative Memory Management
- Rust Gotchas
- Rust Patterns
- Rust Solutions to Rust Problems
- Rust: The Hard Parts
- Strings in Rust
- Thinking in Rust
- Variance in Rust

Please feel welcome to suggest your own headlines and article ideas as well!

The articles I've completed so far:

- [Learning Rust in 2020](https://github.com/pretzelhammer/rust-blog/blob/master/posts/learning-rust-in-2020.md)
- [Common Rust Lifetime Misconceptions](https://github.com/pretzelhammer/rust-blog/blob/master/posts/common-rust-lifetime-misconceptions.md)
- [Sizedness in Rust](https://github.com/pretzelhammer/rust-blog/blob/master/posts/sizedness-in-rust.md)
- [Learn Assembly with Entirely Too Many Brainfuck Compilers](https://github.com/pretzelhammer/rust-blog/blob/master/posts/too-many-brainfuck-compilers.md)

**Comment by lebensterben** [comment]
Rust: Behavior Oriented Programming

**Reply by sistemd** [reply]
Yes

**Comment by towry** [comment]
- Macros in Rust
- Closures in Rust
- Variance in Rust

**Comment by yerke** [comment]
Rust Patterns

**Comment by lmammino** [comment]
Variance in Rust

**Comment by hojjatabdollahi** [comment]
Async Programming in Rust
I would love to see a practical application written with async rust. Something that contains threads, or GUI, or something that makes the combination a little tricky (more than a hello_world.await!

**Reply by pretzelhammer** [reply]
Hi @hojjatabdollahi I wrote a couple articles on async programming Rust:
- [RESTful API in Sync & Async Rust](https://github.com/pretzelhammer/rust-blog/blob/master/posts/restful-api-in-sync-and-async-rust.md) (2021)
- [Beginner's Guide to Concurrent Programming: Building a Multithread Chat Server using Tokio](https://github.com/pretzelhammer/rust-blog/blob/master/posts/chat-server.md) (2024)

I'm not sure if that's what you're looking for though. If you want a deep dive on how async works under-the-hood then this is the best article I know on that subject: [Learn Async Rust with Entirely Too Many Web Servers](https://ibraheem.ca/posts/too-many-web-servers/). Source code for that article [here](https://github.com/ibraheemdev/too-many-web-servers).

**Comment by barkanido** [comment]
Smart pointers, usage of them and usage of their combinations

**Comment by Vagelis-Prokopiou** [comment]
Macros in Rust
Rust: Behavior Oriented Programming
Rust Gotchas
Rust Patterns
Rust Solutions to Rust Problems
Variance in Rust

**Comment by NobodyXu** [comment]
A Spoonful of Sugar Helps the Rust Go Down
Learn Rust with Generative Art
Rust: Behavior Oriented Programming
Rust Gotchas

**Comment by Sphinxes0o0** [comment]
* Rust Patterns
* Async Programming in Rust
* Polymorphism in Rust
* Rust: Behavior Oriented Programming

**Comment by NoblySP** [comment]
- Beginner's Guide to Unsafe Rust
- JavaScript Dev Learns Rust
- Learning Rust in 2021

**Reply by pretzelhammer** [reply]
@NoblySP I recently finished and published [Learning Rust in 2024](https://github.com/pretzelhammer/rust-blog/blob/master/posts/learning-rust-in-2024.md) in case that's something you might be interested in based on your post.

**Comment by rsalmei** [comment]
Most excited to read:
- Thinking in Rust
- Rust Solutions to Rust Problems
- Rust: The Hard Parts
- Rust Patterns

Also grabs my attention:
- Rust Gotchas
- Learn Rust with Generative Art
- A Spoonful of Sugar Helps the Rust Go Down
- Beginner's Guide to Unsafe Rust
- Rust: Behavior Oriented Programming

**Comment by zeeshanakram3** [comment]
- Async Programming in Rust
- Polymorphism in Rust
- Rust: Behavior Oriented Programming

**Comment by abigagli** [comment]
Variance in Rust

**Comment by GNUqb114514** [comment]
Most excited:
- Async Programming in Rust
- Exploring Function Signatures in Rust
- Rust: Behavior Oriented Programming
- Rust: Declarative Memory Management
- Rust Patterns
- Rust Solutions to Rust Problems
- Strings in Rust
- Thinking in Rust
- Macros in Rust

Maybe helpful for others:
- A Spoonful of Sugar Helps the Rust Go Down
- Learning Rust in 2024
- Learn Rust in an Hour
- Learn Rust Real Good Real Fast
- Learn Rust with Generative Art
- Polymorphism in Rust
- Rust Crash Course
- Rust Gotchas
- Rust: The Hard Parts
- Variance in Rust
